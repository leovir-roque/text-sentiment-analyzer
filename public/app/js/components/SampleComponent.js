import React from 'react';

class SampleComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            count: this.props.value,
        }

        this._onClickHandler = this._onClickHandler.bind(this);
    }

    _onClickHandler() {
        this.setState((prevState) => ({
            count: prevState.count + 1,
        }));
    }

    render() {
        return (
            <div className="sample">
                <div className="sample__value">Increment from <strong>{this.state.count}</strong></div>
                <button className="sample__btn" onClick={this._onClickHandler}>Increment</button>
            </div>
        )
    }
}

export default SampleComponent;
