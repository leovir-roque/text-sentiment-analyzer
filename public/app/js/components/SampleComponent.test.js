import React from 'react';
import renderer from 'react-test-renderer';

import SampleComponent from './SampleComponent';

describe('SampleComponent Test', () => {
    test('Renders the component', () => {
        const component = renderer.create(
            <SampleComponent value={0} />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
