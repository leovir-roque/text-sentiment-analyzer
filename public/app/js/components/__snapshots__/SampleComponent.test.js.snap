// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent Test Renders the component 1`] = `
<div
  className="sample"
>
  <div
    className="sample__value"
  >
    Increment from 
    <strong>
      0
    </strong>
  </div>
  <button
    className="sample__btn"
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
